src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -546,8 +546,16 @@ std::string SyncManager::path_for_realm(const SyncConfig& config, util::Optional
 
         // Attempt to make a nicer filename which will ease debugging when
         // locating files in the filesystem.
-        std::string file_name =
-            (custom_file_name) ? custom_file_name.value() : string_from_partition(config.partition_value);
+        auto file_name = [&]() -> std::string {
+            if (custom_file_name) {
+                return *custom_file_name;
+            }
+            if (config.flx_sync_requested) {
+                REALM_ASSERT_DEBUG(config.partition_value.empty());
+                return "flx_sync_default";
+            }
+            return string_from_partition(config.partition_value);
+        }();
         path = m_file_manager->realm_file_path(user->identity(), user->local_identity(), file_name,
```

Comment:
It looks like this code is going to still return the PBS path if the partition is set in the config and the file exists in the filesystem. I think one way to solve this would be to add some config validation (see `RealmCoordinator::set_config`) to make sure that people can't set the partition and flx_sync_requested at the same time.

Reply:
I don't think in practice that this is possible, or at least, it would result in a different error later on because you can't connect to FLX with a PBS realm, but more validation is a good thing. I'm not actually sure where we do validation of the sync config though? It looks like in `RealmCoordinator::set_config()` we validate that the sync config hasn't changed, but don't actually validate that it's correct.

Reply:
We probably just don't have good sync config validation right now 😞  But I think RealmCoordinator::set_config() is probably an appropriate place for it.

Reply:
I see your point, that we would get a sync error down the line which is great, so maybe not a huge deal. But if the config was set to use both that is almost certainly a simple mistake by the developer that can be easily remedied by a clear error message up front.

Reply:
set_config() doesn't check anything on the SyncConfig because there just aren't any invalid combinations of things to check for. If this is introducing one, then set_config() should check for it.